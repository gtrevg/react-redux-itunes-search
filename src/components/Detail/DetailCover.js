import React from 'react';

import './DetailCover.css';

const DetailCover = props => {
	const { cover } = props;
	if (!cover) return null;
	//Trick for improving the size
	const cover300px = cover.replace('100x100bb', '300x300bb');
	return <img alt="Cover not found" src={cover300px} className="DetailCover" />;
};

export default DetailCover;
