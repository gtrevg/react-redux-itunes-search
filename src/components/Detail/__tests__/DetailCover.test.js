import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

import DetailCover from '../DetailCover';

describe('DetailCover component', () => {
	const component = (
		<DetailCover bigCover="http://is2.mzstatic.com/image/thumb/Music127/v4/8a/65/be/8a65bef2-f23d-e43d-9124-f5e4293513f7/source/300x300bb.jpg" />
	);

	it('Renders without crashing', () => {
		const div = document.createElement('div');
		ReactDOM.render(component, div);
		ReactDOM.unmountComponentAtNode(div);
	});

	it('Snapshot matchs', () => {
		const tree = renderer.create(component).toJSON();
		expect(tree).toMatchSnapshot();
	});
});
