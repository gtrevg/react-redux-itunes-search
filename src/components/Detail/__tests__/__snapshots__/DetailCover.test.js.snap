// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailCover component Snapshot matchs 1`] = `
<img
  alt="Cover not found"
  className="DetailCover"
  src="http://is2.mzstatic.com/image/thumb/Music127/v4/8a/65/be/8a65bef2-f23d-e43d-9124-f5e4293513f7/source/300x300bb.jpg"
/>
`;
